index.bs

```diff
@@ -3556,17 +3552,148 @@ A [=remote end=] has a <dfn>before request sent map</dfn> which is initially an
+
+    1. If |url patterns| is [=list/empty=]:
+
+      1. [=list/Append=] |intercept id| to |intercepts|.
+
+      1. Continue.
+
+    1. For each |url pattern| in |url patterns|:
+
+      1. If [=match URL pattern=] with |url pattern| and |url|:
+
+        1. [=list/Append=] |intercept id| to |intercepts|.
+
+        1. Continue.
+
+1. Return |intercepts|.
+
+</div>
+
 ### Types ### {#module-network-types}
 
+#### The network.AuthChallenge Type #### {#type-network-AuthChallenge}
+
+<pre class="cddl local-cddl">
+network.AuthChallenge = {
+  scheme: text,
+  realm: text,
+}
+</pre>
+
+<div algorithm>
+
+To <dfn>extract challenges</dfn> given |response|:
+
+Issue: Should we include parameters other than realm?
+
+1. If |response|'s [=response/status=] is 401, let |header name| be
+   `<code>WWW-Authenticate</code>`. Otherwise if |response|'s [=response/status=] is 407, let
+   |header name| be `<code>Proxy-Authenticate</code>`. Otherwise return null.
+
+1. Let |challenges| be a new [=/list=].
+
+1. For each (|name|, |value|) in |response|'s [=response/header list=]:
+
+   Issue: as in Fetch it's unclear if this is the right way to handle multiple
+   headers, parsing issues, etc.
+
+  1. If |name| is a [=byte-case-insensitive=] match for |header name|:
+
+    1. Let |header challenges| be the result of parsing |value| into a list of
+       challenges, each consisting of a scheme and a list of parameters, each of
+       which is a [=tuple=] (name, value), according to the rules of [[!RFC9110]].
+
+    1. For each |header challenge| in |header challenges|:
+
+      1. Let |scheme| be |header challenge|'s scheme.
+
+      1. Let |realm| be the empty string.
+
+      1. For each (|param name|, |param value|) in |header challenge|'s
+         parameters:
+            1. If |param name| equals `<code>realm</code>` let |realm| be
+               [=UTF-8 decode=] |param value|.
+
+      1. Let |challenge| be a new [=/map=] matching the
+         <code>network.AuthChallenge</code> production, with the
+         <code>scheme</code> field set to |scheme| and the <code>realm</code>
+         field set to |realm|.
+
+     1. [=list/Append=] |challenge| to |challenges|.
+
+1. Return |challenges|.
+
+</div>
+
+#### The network.AuthCredentials Type #### {#type-network-AuthCredentials}
+
+<pre class="cddl remote-cddl">
+network.AuthCredentials = {
+  type: "password",
+  username: text,
+  password: text,
+}
+</pre>
+
+The <code>network.AuthCredentials</code> type represents the response to a
+request for authorization credentials.
+
 #### The network.BaseParameters Type #### {#type-network-BaseParameters}
 
 <pre class="cddl local-cddl">
 network.BaseParameters = (
     context: browsingContext.BrowsingContext / null,
+    isBlocked: bool,
     navigation: browsingContext.Navigation / null,
     redirectCount: js-uint,
     request: network.RequestData,
     timestamp: js-uint,
+    ? intercepts: [*network.Intercept]
```

Comment:
```suggestion
    ? intercepts: [+network.Intercept]
```

By definition intercepts cannot be empty? If it's empty, then isBlocked is true and then `intercepts` should be omitted?